### DISCUSSION TITLE
Traefik "Gateway Timeout"

### DISCUSSION THREAD

**Discussion by EZPC-Matt** [dani-garcia/vaultwarden#5172] [Q&A]
I have an issue with VW being traefik where after it's been running for a while I start getting gateway timeout errors.

Restarting VW fixes this, I'm wondering if something is triggering VW to temporarily block the traefik reverse proxy as it seems VW itself is still working (Curling the containers IP directly from the host works)

**Comment by EZPC-Matt** [comment]
If anyone else runs into this I think it was a case of Traefik trying to use the wrong network to communicate with the container.

The VW container had 2 docker networks, one it shared with the frontend traefik proxy and a backend network so it can talk to the DB server, it seems after some time traefik was switching to the IP for the internal (VW) network for some reason and was no longer able to reach VW as traefik itself wasn't present on that network.

Add a label to docker-compose as follows

`"traefik.docker.network=traefik-network"`

Replace "Traefik-network" with whatever you called the docker network both traefk and the VW container sit on.

**Reply by faceslog** [reply]
ty man
